caps/management/commands/process_scottish_reporting_data.py

```diff
@@ -0,0 +1,428 @@
+import re
+from collections import defaultdict
+
+from os.path import join
+
+import pandas as pd
+import numpy as np
+
+from django.core.management.base import BaseCommand
+from django.conf import settings
+
+
+class Command(BaseCommand):
+    help = "processing scottish council climate data"
+
+    data = defaultdict(list)
+
+    report = None
+    sheet = None
+    sheet_name = None
+    report_data = None
+    data_type = None
+
+    subsets = {
+        "council_data": {
+            "desc": "basic council data",
+            "method": "extract_council_data_from_sheet",
+        },
+        "emissions": {
+            "desc": "emissions data",
+            "method": "extract_emissions_data_from_sheet",
+            "header_text": "Baseline Year",
+            "start_text": "Reference year",
+        },
+        "sources": {
+            "desc": "emissions sources data",
+            "method": "extract_emissions_sources_from_sheet",
+            "header_text": "Emission source",
+        },
+        "targets": {
+            "desc": "reduction targets data",
+            "method": "extract_targets_from_sheet",
+            "header_text": r"Name of [Tt]arget",
+        },
+        "projects": {
+            "desc": "emission reduction projects data",
+            "method": "extract_projects_from_sheet",
+            "header_text": "Project name",
+        },
+        "generation": {
+            "desc": "renewables generation data",
+            "method": "extract_generation_from_sheet",
+            "header_text": "Generation, consumption and export of renewable energy",
+            "start_text": "Technology",
+        },
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--all",
+            default=True,
+            action="store_true",
+            help="Update all data",
+        )
+
+        parser.add_argument(
+            "--sample",
+            action="store_true",
+            help="Just process the first 20 rows for testing",
+        )
+
+        for arg, desc in self.subsets.items():
+            parser.add_argument(
+                "--{}".format(arg),
+                action="store_true",
+                help="Update {}".format(desc),
+            )
+
+    def handle(self, *args, **options):
+        self.options = options
+
+        get_all = options["all"]
+        for option in self.subsets.keys():
+            if options[option]:
+                self.options["all"] = False
+                break
+
+        self.process_data()
+        self.save_data()
+
+    def process_data(self):
+        data = join(settings.SCOTTISH_DIR, "data_list.csv")
+        df = pd.read_csv(data)
+
+        for index, row in df.iterrows():
+            try:
+                self.report_data = row
+                self.process_report()
+                if self.options["sample"] and index == 20:
```

Comment:
[comment, it's fine] this could also be
```
pd.read_csv(data, limit = 20 if self.options["sample"] else None)
```

or a less annoying way of setting the number